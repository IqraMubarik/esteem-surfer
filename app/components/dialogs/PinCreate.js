import React, {Component} from 'react';


import PropTypes from 'prop-types';


import {Modal} from 'antd';
import {FormattedHTMLMessage, FormattedMessage, injectIntl} from 'react-intl';


class PinCreate extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {

  }



  render() {
    const {global} = this.props;


    return (
      null
    );
  }
}

PinCreate.propTypes = {
  global: PropTypes.shape({}).isRequired
};

export default injectIntl(PinCreate);
