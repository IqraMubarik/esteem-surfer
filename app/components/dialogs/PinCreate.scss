@import '../../style/mixins';
@import '../../style/colors';

.pin-create-dialog-content {
  padding: 35px;

  .dialog-form {
    align-items: center;
    display: flex;
    flex-wrap: wrap;

    .brand {
      height: 128px;
      margin-right: 12px;
      width: 128px;

      img {
        width: 100%;
      }
    }

    .form-content {
      padding-top: 14px;
      width: calc(100% - 140px);
    }

    .form-content {
      .form-title {
      }

      .form-desc {
        color: $steel-grey;
      }

      .form-desc2 {
        color: $silver;
        margin: 1em 0 0 0;
      }
    }
  }
}
