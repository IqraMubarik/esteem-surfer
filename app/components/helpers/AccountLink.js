import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getAccount } from '../../backend/steem-client';

export const makePath = username => `/@${username}`;

class AccountLink extends Component {
  goProfile = async () => {
    const { username, history, actions } = this.props;
    let { accountData } = this.props;
    const { setVisitingAccount } = actions;

    if (!accountData) {
      try {
        accountData = await getAccount(username);
      } catch (err) {
        accountData = null;
      }
    }

    if (accountData) {
      setVisitingAccount(accountData);
    }

    history.push(makePath(username));
  };

  render() {
    const { children } = this.props;

    return React.cloneElement(children, {
      onClick: this.goProfile
    });
  }
}

AccountLink.defaultProps = {
  accountData: null
};

AccountLink.propTypes = {
  children: PropTypes.element.isRequired,
  username: PropTypes.string.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  accountData: PropTypes.instanceOf(Object),
  actions: PropTypes.shape({
    setVisitingAccount: PropTypes.func.isRequired
  }).isRequired
};

export default AccountLink;
