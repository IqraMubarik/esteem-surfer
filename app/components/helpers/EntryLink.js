import React, {Component} from 'react';
import PropTypes from 'prop-types';

class EntryLink extends Component {
  goEntry = () => {
    const {entry, history, actions} = this.props;

    const {category, author, permlink} = entry;

    actions.setVisitingEntry(entry);

    history.push(`/${category}/@${author}/${permlink}`);
  };

  render() {
    const {children} = this.props;

    return React.cloneElement(children, {
      onClick: this.goEntry
    });
  }
}

EntryLink.defaultProps = {};

EntryLink.propTypes = {
  children: PropTypes.element.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  entry: PropTypes.shape({
    category: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    permlink: PropTypes.string.isRequired
  }).isRequired,
  actions: PropTypes.shape({
    setVisitingEntry: PropTypes.func.isRequired
  }).isRequired
};

export default EntryLink;
