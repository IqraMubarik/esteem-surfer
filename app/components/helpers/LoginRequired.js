import React, {Component, Fragment} from 'react';
import PropTypes from "prop-types";
import Login from '../dialogs/Login'
import {Modal} from "antd";


class LoginRequired extends Component {


  constructor(props) {
    super(props);

    this.state = {
      loginModalVisible: false
    };
  }


  showLoginModal = () => {

    console.log("aaa")
    this.setState({
      loginModalVisible: true
    });
  };

  onLoginModalCancel = () => {
    this.setState({
      loginModalVisible: false
    });
  };

  onLoginSuccess = () => {
    this.setState({
      loginModalVisible: false
    });
  };

  render() {
    const {children, activeAccount, requiredKeys} = this.props;
    const {loginModalVisible} = this.state;

    if (activeAccount) {
      return children
    }


    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.showLoginModal()
      }
    });

    return (
      <Fragment>
        {newChild}

        <Modal
          visible={loginModalVisible}
          onCancel={this.onLoginModalCancel}
          footer={false}
          width="500px"
          closable={false}
          destroyOnClose
          centered
        >
          <Login {...this.props} onSuccess={this.onLoginSuccess}/>
        </Modal>
      </Fragment>
    )
  }
}

LoginRequired.defaultProps = {
  activeAccount: null,
};


LoginRequired.propTypes = {
  children: PropTypes.element.isRequired,
  activeAccount: PropTypes.instanceOf(Object),
  requiredKeys: PropTypes.array.isRequired
};

export default LoginRequired;
