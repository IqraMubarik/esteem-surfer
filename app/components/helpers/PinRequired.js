import React, { Component, Fragment } from 'react';
import { Modal } from 'antd';

import PropTypes from 'prop-types';
import PinConfirm from '../dialogs/PinConfirm';


class PinRequired extends Component {

  constructor(props) {
    super(props);

    this.state = {
      show: false
    };
  }

  onPinInvalidated = () => {

  };

  onConfirmPinSuccess = () => {
    const { children } = this.props;
    const { onClick } = children.props;
    this.setState({ show: false }, () => {
      onClick();
    });
  };

  render() {
    const { children } = this.props;
    const { show } = this.state;

    // Clone child element rewriting onclick event
    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.setState({ show: true });
      }
    });

    return (
      <Fragment>
        {show &&
        <Modal
          footer={null}
          closable={false}
          keyboard={false}
          visible
          width="500px"
          centered
          destroyOnClose
        >
          <PinConfirm
            {...this.props}
            onSuccess={this.onConfirmPinSuccess}
            onInvalidate={this.onPinInvalidated}
          />
        </Modal>
        }
        {newChild}
      </Fragment>
    );
  }
}

PinRequired.defaultProps = {};

PinRequired.propTypes = {
  children: PropTypes.element.isRequired
};

export default PinRequired;
