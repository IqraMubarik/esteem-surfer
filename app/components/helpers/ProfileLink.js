import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getAccount } from '../../backend/steem-client';

class ProfileLink extends Component {
  goProfile = async () => {
    const { username, history, actions } = this.props;
    let { accountData } = this.props;
    const { setVisitingProfile } = actions;

    if (!accountData) {
      try {
        accountData = await getAccount(username);
      } catch (err) {
        accountData = null;
      }
    }

    if (accountData) {
      setVisitingProfile(accountData);
    }

    history.push(`/@${username}`);
  };

  render() {
    const { children } = this.props;

    return React.cloneElement(children, {
      onClick: this.goProfile
    });
  }
}

ProfileLink.defaultProps = {
  accountData: null
};

ProfileLink.propTypes = {
  children: PropTypes.element.isRequired,
  username: PropTypes.string.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  accountData: PropTypes.instanceOf(Object),
  actions: PropTypes.shape({
    setVisitingProfile: PropTypes.func.isRequired
  }).isRequired
};

export default ProfileLink;
