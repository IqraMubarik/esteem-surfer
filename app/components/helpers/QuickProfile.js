/*
eslint-disable import/no-cycle
*/

import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Drawer, Button, Tooltip, message } from 'antd';
import { FormattedMessage, FormattedNumber, injectIntl } from 'react-intl';

import {
  getAccounts,
  getDiscussions,
  getFollowCount
} from '../../backend/steem-client';
import authorReputation from '../../utils/author-reputation';

import UserAvatar from '../elements/UserAvatar';
import EntryListLoadingItem from '../elements/EntryListLoadingItem';
import LinearProgress from '../common/LinearProgress';
import EntryListItem from '../elements/EntryListItem';

class QuickProfile extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false,
      active: false,
      profile: {
        name: ' ',
        about: ' ',
        postCount: 0
      },
      follows: {
        followerCount: null,
        followingCount: null
      },
      loadingEntries: true,
      entries: []
    };
  }

  load = async () => {
    const { username, intl } = this.props;

    const accounts = await getAccounts([username]);
    const account = accounts[0];

    let accountProfile;
    try {
      accountProfile = JSON.parse(account.json_metadata).profile;
    } catch (err) {
      message.error(
        intl.formatMessage({ id: 'quick-profile.fetch-error-profile' })
      );
    }

    if (accountProfile) {
      const name = accountProfile.name || null;
      const about = accountProfile.about || null;
      const postCount = account.post_count;

      this.setState({ profile: { name, about, postCount } });
    }

    let follow;
    try {
      follow = await getFollowCount(username);
    } catch (err) {
      message.error(
        intl.formatMessage({ id: 'quick-profile.fetch-error-profile' })
      );
    }

    if (follow) {
      const followerCount = follow.follower_count;
      const followingCount = follow.following_count;

      this.setState({ follows: { followerCount, followingCount } });
    }

    let entries;
    try {
      entries = await getDiscussions('blog', {
        tag: username,
        limit: 7,
        start_author: undefined,
        start_permlink: undefined
      });
    } catch (err) {
      message.error(
        intl.formatMessage({ id: 'quick-profile.fetch-error-content' })
      );
    }

    if (entries) {
      this.setState({ entries, loadingEntries: false });
    }
  };

  show = () => {
    this.setState({ active: true, visible: true });
    this.load();
  };

  hide = () => {
    this.setState({ visible: false });
    setTimeout(() => {
      this.setState({ active: false });
    }, 500);
  };

  render() {
    const { children, username, reputation } = this.props;
    const {
      visible,
      active,
      profile,
      follows,
      entries,
      loadingEntries
    } = this.state;

    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.show();
      }
    });

    const followersMsg =
      follows.followerCount === null ? (
        '--'
      ) : (
        <FormattedNumber value={follows.followerCount} />
      );
    const postCountMsg =
      profile.postCount === null ? (
        '--'
      ) : (
        <FormattedNumber value={profile.postCount} />
      );
    const followingMsg =
      follows.followingCount === null ? (
        '--'
      ) : (
        <FormattedNumber value={follows.followingCount} />
      );

    return (
      <Fragment>
        {newChild}

        {active && (
          <Drawer
            placement="right"
            closable={false}
            onClose={this.hide}
            visible={visible}
            width="640px"
          >
            <div
              className={`quick-profile-content ${
                loadingEntries ? 'loading' : ''
              } `}
            >
              <div className="profile-area">
                <div className="follow-btn-holder">
                  <Tooltip placement="left" title={`Follow @${username}`}>
                    <Button type="primary" shape="circle" size="large">
                      <i className="mi">person_add</i>
                    </Button>
                  </Tooltip>
                </div>
                <div className="profile-avatar">
                  <UserAvatar user={username} size="large" />
                  <div className="reputation">
                    {authorReputation(reputation)}
                  </div>
                </div>
                {profile.name && (
                  <div className="full-name">{profile.name}</div>
                )}
                <div className="username">{username}</div>
                {profile.about && <div className="about">{profile.about}</div>}
                <div className="numbers">
                  <span className="followers">
                    <FormattedMessage
                      id="quick-profile.n-followers"
                      values={{ n: followersMsg }}
                    />
                  </span>
                  <span className="post-count">
                    <FormattedMessage
                      id="quick-profile.n-posts"
                      values={{ n: postCountMsg }}
                    />
                  </span>
                  <span className="following">
                    <FormattedMessage
                      id="quick-profile.n-following"
                      values={{ n: followingMsg }}
                    />
                  </span>
                </div>
              </div>

              {loadingEntries && (
                <Fragment>
                  <LinearProgress />
                  <div className="entries">
                    <EntryListLoadingItem />
                  </div>
                </Fragment>
              )}

              {!loadingEntries && (
                <div className="entries">
                  {entries.map(d => (
                    <EntryListItem
                      key={d.id}
                      {...this.props}
                      entry={d}
                      inDrawer
                    />
                  ))}
                </div>
              )}
            </div>
          </Drawer>
        )}
      </Fragment>
    );
  }
}

QuickProfile.defaultProps = {};

QuickProfile.propTypes = {
  children: PropTypes.element.isRequired,
  username: PropTypes.string.isRequired,
  reputation: PropTypes.string.isRequired
};

export default injectIntl(QuickProfile);
