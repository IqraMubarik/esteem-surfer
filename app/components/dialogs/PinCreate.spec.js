/* eslint-disable */

import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PinCreate from './PinCreate';
import { mountWithIntl } from '../../helpers/intl-test';

Enzyme.configure({ adapter: new Adapter() });

describe('PinCreate', () => {
  it('(1) Render', () => {
    const props = {
      actions: {
        exposePin: () => {},
        wipePin: () => {}
      }
    };

    const wrapper = mountWithIntl(
      <div>
        <PinCreate {...props} />
      </div>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
