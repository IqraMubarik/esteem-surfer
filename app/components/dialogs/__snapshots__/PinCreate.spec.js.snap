// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinCreate (1) Render 1`] = `"<div intl=\\"[object Object]\\"></div>"`;
